docs/assets.rst

```diff
@@ -0,0 +1,673 @@
+These are built with ``make localecompile`` when creating the production
+images or when a developer wants to see updated translations.
+
+.. _Pontoon: https://pontoon.mozilla.org/projects/mdn/
+.. _mdn-l10n: https://github.com/mozilla-l10n/mdn-l10n
+.. _`git submodule`: https://github.blog/2016-02-01-working-with-submodules/
+.. _gettext: https://en.wikipedia.org/wiki/Gettext
+.. _Puente: https://puente.readthedocs.io/en/latest/
+
+.. _build-locale-js:
+
+Building localization JavaScript
+================================
+Django includes a `JavaScriptCatalog view`_ that provides JavaScript
+implementations of gettext_ functions, as well as translations for each
+locale. It is ineffecient to use this view directly, since it is generated
+on access. For efficiency, django-statici18n_ generates files for each locale
+from the ``JavaScriptCatalog`` output, so they can be served as static assets.
+
+The translation catalog files are created with ``make compilejsi18n``
+from the locale Machine Object ``.mo`` files.  Kuma sets ``STATICI18N_ROOT``
+to ``build/locale``, and the output files have names like
+``build/locale/jsi18n/de/javascript.js``.
+
+.. _`JavaScriptCatalog view`: https://docs.djangoproject.com/en/1.11/topics/i18n/translation/#module-django.views.i18n
+.. _django-statici18n: https://django-statici18n.readthedocs.io/en/latest/
+
+.. _build-ckeditor:
+
+Building CKEditor
+=================
+CKEditor is a complex JavaScript application that provides a WYSIWYG editor
+for MDN wiki pages. It is packaged with plugins, some from third parties,
+and some custom to MDN.
+
+The CKEditor build process is documented on the
+:doc:`CKEditor document</ckeditor>`. The built files are checked into the
+Kuma repository.
+
+.. _include-js-libraries:
+
+Including JS libraries
+======================
+Third-party JavaScript libraries are included in the Kuma repository.
+Some were added manually, and others with Bower_. See
+:ref:`front-end-asset-dependencies` for more details about these
+libraries.
+
+Some of these libraries are served directly to visitors, while others are
+included in pipleline JavaScript bundles.
+
+.. _Bower: http://bower.io
+
+.. _pipeline-js-bundles:
+
+Building pipeline JavaScript bundles
+====================================
+Pipeline JavaScript bundles combine several JavaScript files into a single
+file, with optional minimization. For example, the file
+`static/build/js/main.js`_ is the combination of 10 JavaScript files:
+
+* `kuma/static/js/libs/jquery/jquery.js`_ (JQuery 2.2.0)
+* `kuma/static/js/libs/icons.js`_
+* `kuma/static/js/components.js`_
+* `kuma/static/js/analytics.js`_
+* `kuma/static/js/main.js`_
+* `kuma/static/js/components/nav-main-search.js`_
+* `kuma/static/js/auth.js`_
+* `kuma/static/js/highlight.js`_
+* `kuma/static/js/wiki-compat-trigger.js`_
+* `kuma/static/js/lang-switcher.js`_
+
+In development, the source files (10 for ``main.js``) are served, so there are
```

Comment:
Is it worth specifying here (perhaps with just a link) how the build process knows whether it is in development or production? As I read this that is a detail that I do not remember.

Reply:
The short answer is ``DEBUG=True`` or ``=False``, but it is quite a bit more complicated, because ``django-pipeline`` is too much magic. I'll move the links higher in the paragraph.